packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -141,22 +157,32 @@ export class State {
   }
 
   private toStateJson(scope: Construct, actionBindOptions: ActionBindOptions): CfnDetectorModel.StateProperty {
-    const { onEnter } = this.props;
+    const { onEnter, onInput, onExit } = this.props;
     return {
       stateName: this.stateName,
-      onEnter: onEnter && { events: toEventsJson(scope, actionBindOptions, onEnter) },
```

Comment:
Why are we getting rid of the `onEnter` condition? This means this will render an empty `{ "OnEnter": {} }` objects in the template.

Can we keep this condition? In fact, add it to `onInput ` and `onExit`?